includes/init.php

```diff
@@ -112,6 +112,43 @@ function pmpro_init()
 }
 add_action("init", "pmpro_init");
 
+/**
+ * Add filter for timeout
+ *
+ * Change GET/POST
+ */
+add_action( 'wp_enqueue_scripts', 'pmpro_checkout_scripts' );
+function pmpro_checkout_scripts() {
+	global $pmpro_pages;
+	wp_register_script( 'checkout-page', plugins_url( '/js/checkout-page.js', __DIR__ ), array( 'jquery' ), '1.2' );
+	wp_localize_script(
+		'checkout-page',
+		'checkout_page_object',
+		array(
+			'checkout_page_ajaxurl'   => admin_url( 'admin-ajax.php' ),
+			'checkout_page_nonce'     => wp_create_nonce( 'checkout-page-nonce' ),
+			'applydiscountcode'		  => apply_filters( 'pmpro_ajax_timeout', 5000, 'applydiscountcode' ),
+		)
+	);
+	wp_register_style( 'checkout-page', plugins_url( '/css/checkout-page.css', __DIR__ ),  time() );
+	if ( is_page( $pmpro_pages['checkout'] ) ) {
```

Comment:
This is unreliable since you can have the checkout shortcode on multiple pages.

Instead, let's enqueue the script from within preheaders/checkout.php, which should work and will make sure that the code is only enqueued on checkout pages.

Reply:
Fair enough. We can register here and enqueue from `preheaders`